### ISSUE TITLE
nokogiri requires Ruby version < 3.1.dev, >= 2.5

### ISSUE THREAD

**Issue by mupeifeiyi** [vagrant-libvirt/vagrant-libvirt#1214]
When i am executing `vagrant plugin install vagrant-libvirt `
The error output is as follows
```
Installing the 'vagrant-libvirt' plugin. This can take a few minutes...
Fetching: excon-0.79.0.gem (100%)
Fetching: formatador-0.2.5.gem (100%)
Fetching: fog-core-2.2.3.gem (100%)
Fetching: fog-json-1.2.0.gem (100%)
Fetching: racc-1.5.2.gem (100%)
Building native extensions.  This could take a while...
Bundler, the underlying system Vagrant uses to install plugins,
reported an error. The error is shown below. These errors are usually
caused by misconfigured plugin installations or transient network
issues. The error from Bundler is:

nokogiri requires Ruby version < 3.1.dev, >= 2.5.
```

Actually I have upgraded ruby, the above error is still output。

**Comment by cstroe** [comment]
After upgrading to Vagrant version 2.2.14 (latest as of this writing), the installation of vagrant-libvirt succeeded:

```
$ vagrant version
Installed Version: 2.2.14
Latest Version: 2.2.14
 
You're running an up-to-date version of Vagrant!
$ vagrant plugin install vagrant-libvirt
Installing the 'vagrant-libvirt' plugin. This can take a few minutes...
Fetching formatador-0.2.5.gem
Fetching fog-core-2.2.3.gem
Fetching fog-json-1.2.0.gem
Fetching racc-1.5.2.gem
Building native extensions. This could take a while...
Fetching nokogiri-1.11.1-x86_64-linux.gem
Fetching fog-xml-0.1.3.gem
Fetching ruby-libvirt-0.7.1.gem
Building native extensions. This could take a while...
Fetching fog-libvirt-0.8.0.gem
Fetching vagrant-libvirt-0.3.0.gem
Installed the plugin 'vagrant-libvirt (0.3.0)'!
```

**Comment by mupeifeiyi** [comment]
> 升级到Vagrant 2.2.14版（撰写本文时为最新）之后，成功安装了vagrant-libvirt：
> 
> ```
> $ wget https://releases.hashicorp.com/vagrant/2.2.14/vagrant_2.2.14_x86_64.deb
> ...
> $ sudo dpkg -i vagrant_2.2.14_x86_64.deb 
> (Reading database ... 191136 files and directories currently installed.)
> Preparing to unpack vagrant_2.2.14_x86_64.deb ...
> Unpacking vagrant (1:2.2.14) over (1:2.2.6) ...
> Setting up vagrant (1:2.2.14) ...
> $ vagrant version
> Installed Version: 2.2.14
> Latest Version: 2.2.14
>  
> You're running an up-to-date version of Vagrant!
> $ vagrant plugin install vagrant-libvirt
> Installing the 'vagrant-libvirt' plugin. This can take a few minutes...
> Fetching formatador-0.2.5.gem
> Fetching fog-core-2.2.3.gem
> Fetching fog-json-1.2.0.gem
> Fetching racc-1.5.2.gem
> Building native extensions. This could take a while...
> Fetching nokogiri-1.11.1-x86_64-linux.gem
> Fetching fog-xml-0.1.3.gem
> Fetching ruby-libvirt-0.7.1.gem
> Building native extensions. This could take a while...
> Fetching fog-libvirt-0.8.0.gem
> Fetching vagrant-libvirt-0.3.0.gem
> Installed the plugin 'vagrant-libvirt (0.3.0)'!
> ```

ok，Thanks，This is very helpful for me

**Comment by electrofelix** [comment]
Looks like nokogiri has removed support for older rubies, though I'm a little stumped as to why the embedded bundler is not detecting the correct version of nokogiri based on the dependencies, since this works for the unit tests which is using native bundler to install against older ruby versions. Suspect there may be a bug with vagrant itself so I'll leave this open until I log a replacement issue upstream.

**Comment by electrofelix** [comment]
I'm going to assume that the original error here was related to an older vagrant. I wasn't able to track down what was fixed that allowed 2.2.14 to fix the problem, but I guess since vagrant was fixed there is no issue to open upstream.

Closing for now unless someone else bumps into this.

**Comment by nttg8100** [comment]
I meet the similar issue. I tried to use with vagrant 2.2.14 and 2.3.4 and 2.4.1 but there is no luck for me.
